spyder/config/main.py

```diff
@@ -207,6 +207,7 @@
               'automatic_completions': True,
               'automatic_completions_after_chars': 3,
               'automatic_completions_after_ms': 300,
+              'completions_max_request_ms': 200,

```

Comment:
Sometimes Jedi takes long to complete some compiled packages such as numpy, which Kite returns faster. What happens if just LSP is working on such cases? I think that this value is small when LSP is the only provider available, besides, it could depend on the actual speed and load of the machine running Spyder.

Reply:
> What happens if just LSP is working on such cases?

This setting acts as a ceiling on the amount of time completions requests are allowed to take. So if LSP is the only provider here, the request will time out and the user will see no completions.

> I think that this value is small when LSP is the only provider available

If we think that this default value isn't giving LSP enough time to return, then we can increase the default value to something like 500ms. In any case, the point of a setting is to let the user change it according to their personal tastes or based on the specs of their machine. The default value is just a best guess at what should be a reasonable number.

Reply:
@its-dhung, this is a solution we considered before with @andfoy but we found problematic because Jedi could take up to 3 or 4 seconds to return its first completion, if it involves a compiled library (e.g. Numpy or Pandas). Besides, it also takes a long time (2 secs) for big classes (e.g. our own MainWindow one).

Therefore, we think this solution is unfeasible because it'd rule out Jedi in several important cases.

Reply:
I've included a change to allow the user to disable this functionality by setting the value to 0.

Reply:
I think the right solution based on the feedback thus far is:

1. Before the timeout, continue waiting if not all `WAIT_FOR` sources have returned.
2. After the timeout, return once any `WAIT_FOR` source has returned a non-empty response (the same as the current behavior).

So, if only LSP is enabled, and LSP takes longer than 200ms, then we'd continue waiting for LSP to return due to criterion 2.

Do you agree with this? @ccordoba12 @andfoy

Reply:
I'm going to open a separate PR, because this branch is on @its-dhung's personal GH, so I can't modify it.